dataless-classifier/src/main/java/edu/illinois/cs/cogcomp/datalessclassification/representation/w2v/MemoryBasedW2V.java

```diff
@@ -0,0 +1,230 @@
+ * is thus quite fast.
+ * 
+ * @author yqsong@illinois.edu
+ * @author shashank
+ */
+
+public class MemoryBasedW2V extends AEmbedding<Integer> {
+    private static Logger logger = LoggerFactory.getLogger(MemoryBasedW2V.class);
+
+    private static Map<String, DenseVector> vectors;
+
+    private int dimensions;
+
+    public MemoryBasedW2V() {
+        this(new W2VDatalessConfigurator().getDefaultConfig());
+    }
+
+    public MemoryBasedW2V(ResourceManager config) {
+        this(config.getInt(W2VDatalessConfigurator.W2V_DIM));
+    }
+
+    public MemoryBasedW2V(int embSize) {
+        dimensions = embSize;
+    }
+
+    public void loadVectors() {
+        if (vectors == null) {
+            try {
+                File inputFile = getFile();
+                logger.info("Reading Word2vec Embeddings from " + inputFile.getAbsolutePath());
+                vectors = new HashMap<>();
+
+                BufferedReader bf = new BufferedReader(new FileReader(inputFile));
+
+                String line = bf.readLine();
+                String[] tokens = line.split(" ");
+
+                // The first line has the following schema --> #Terms #Vector_Dimensions
+                int dimNum = Integer.parseInt(tokens[1].trim());
+
+                if (dimNum != dimensions) {
+                    bf.close();
+                    throw new Exception("Number of dimensions in the embeddings file (" + dimNum
+                            + ") don't match the one in the config file (" + dimensions + ")");
+                }
+
+                int count = 0;
+
+                while ((line = bf.readLine()) != null) {
+                    line = line.trim();
+
+                    if (line.length() == 0)
+                        continue;
+
+                    tokens = line.trim().split(" ", 2);
+                    String[] stringVec = tokens[1].split(" ");
+
+                    if (stringVec.length != dimNum) {
+                        bf.close();
+                        throw new Exception(
+                                "Possible Error in the embeddings file -- number of dimensions("
+                                        + dimNum + ") don't match -->" + tokens[1]);
+                    }
+
+                    String word = tokens[0].trim();
+                    if (word.length() == 0)
+                        continue;
+
+                    double[] scores = new double[dimNum];
+
+                    int i = 0;
+                    for (String dim : stringVec) {
+                        scores[i] = Double.parseDouble(dim);
+                        i++;
+                    }
+
+                    DenseVector vec = new DenseVector(scores);
+                    vectors.put(word, vec);
+
+                    count++;
+
+                    if (count % 100000 == 0)
+                        logger.info("#W2V embeddings read: " + count);
+
+                }
+
+                bf.close();
```

Comment:
as before -- close in 'finally' block or use try-with-resources

Reply:
Done.